test/tester.go

```diff
@@ -319,17 +288,244 @@ func (at *AccountsTester) HealthGet() (api.HealthGET, int, error) {
 	return resp, r.StatusCode, nil
 }
 
-// UserAPIKeysDELETE performs a `DELETE /user/apikeys/:id` request.
+/*** Login and logout helpers ***/
+
+// LoginCredentialsPOST logs the user in and returns a response.
+//
+// NOTE: The Body of the returned response is already read and closed.
+func (at *AccountsTester) LoginCredentialsPOST(emailAddr, password string) (*http.Response, []byte, error) {
+	params := url.Values{}
+	params.Set("email", emailAddr)
+	params.Set("password", password)
+	return at.post("/login", nil, params)
+}
+
+// LoginPubKeyGET performs `GET /login`
+func (at *AccountsTester) LoginPubKeyGET(pk database.PubKey) (api.ChallengePublic, int, error) {
+	params := url.Values{}
+	if pk != nil {
+		params.Set("pubKey", hex.EncodeToString(pk[:]))
+	}
+	r, b, err := at.Request(http.MethodGet, "/login", params, nil, nil)
```

Comment:
If you feel fancy you can add wrappers for post, get and delete to avoid the `http.MethodGet`, `http.MethodPost` etc.

Reply:
I don't think those will bring enough value at the cost of adding another level of abstraction.

I was considering hiding the entire response processing within `Request` but it seemed like too much complexity with all moving parts such as expected status codes and so on. I might put in there the body processing, though.